scripts/synthetix/facilitate_trade.js

```diff
@@ -73,44 +88,89 @@ module.exports = async (callback) => {
+      sETH.exchangeId,
+      sUSD.exchangeId,
+      floatToErc20Units(MIN_SELL_USD, sUSD.decimals),
+      networkId
+    )
+    console.log("Exchange buy  sETH price (in sUSD)", 1 / exchangeBuyETHPrice)
+
+    const minSellETH = floatToErc20Units(MIN_SELL_USD / formatedRate, sETH.decimals)
+    const exchangeSellETHPrice = await estimatePrice(sUSD.exchangeId, sETH.exchangeId, minSellETH, networkId)
+    console.log("Exchange sell sETH price (in sUSD)", exchangeSellETHPrice)
 
     // Using synthetix's fees, and formatting their return values with their tools, plus parseFloat.
     const sETHTosUSDFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sETHKey, sUSDKey)))
     const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
 
-    // Compute buy-sell amounts based on unlimited orders with rates from above.
-    const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
-      1 / (formatedRate * (1 - sUSDTosETHFee)),
-      sETH.decimals,
-      sUSD.decimals
-    )
-    const { base: sellETHAmount, quote: buySUSDAmount } = getUnlimitedOrderAmounts(
-      formatedRate * (1 + sETHTosUSDFee),
-      sUSD.decimals,
-      sETH.decimals
-    )
-
-    const buyAmounts = [buyETHAmount, buySUSDAmount]
-    const sellAmounts = [sellSUSDAmount, sellETHAmount]
-
-    // Fetch auction index and declare validity interval for orders.
-    // Note that order validity interval is inclusive on both sides.
-    const batchId = (await exchange.getCurrentBatchId.call()).toNumber()
-    const validFroms = Array(2).fill(batchId)
-    const validTos = Array(2).fill(batchId)
-
-    // Avoid querying exchange by tokenAddress for fixed tokenId
-    const buyTokens = [sETH, sUSD].map((token) => token.exchangeId)
-    const sellTokens = [sUSD, sETH].map((token) => token.exchangeId)
-
-    const gasPrices = await (await fetch(gasStationURL[networkId])).json()
-    const scaledGasPrice = parseInt(gasPrices[argv.gasPrice] * argv.gasPriceScale)
-    console.log(`Using current "${argv.gasPrice}" gas price scaled by ${argv.gasPriceScale}: ${scaledGasPrice}`)
-    await exchange.placeValidFromOrders(buyTokens, sellTokens, validFroms, validTos, buyAmounts, sellAmounts, {
-      from: account,
-      gasPrice: scaledGasPrice,
-    })
+    // Initialize order array.
+    const orders = []
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above when the price is right.
+    const ourBuyETHRate = formatedRate * (1 - sUSDTosETHFee)
+    if (ourBuyETHRate > 1 / exchangeBuyETHPrice) {
+      // If we are willing to pay more than the exchange quotes as buy price.
+      console.log(`Placing an order to buy sETH at ${ourBuyETHRate}`)
+      const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
+        1 / ourBuyETHRate,
+        sETH.decimals,
+        sUSD.decimals
```

Comment:
Not an issue of this PR, but I wonder if it makes sense to have the input to this function be an object, so you have to specify `quoteTokenDecimal: sETH.decimals` and so on. You'd see immediately if there had been mismatches.

Reply:
This is certainly something I would like to tackle and it has been mentioned (in passing) during the TS migration of `dex-contracts`. Funny that this actually used to be the case, but I had requested it be removed since only the decimals attribute of the token object was ever used. 

In reality, from an object oriented programming perspective `Token` should be a class and this method should be implemented on that class, then it would be `token.getUnlimitedOrderAmounts(price, otherToken)`